sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/KafkaBundle.java

```diff
@@ -212,9 +222,12 @@ public <K, V> MessageProducer<K, V> registerProducer(ProducerRegistration<K, V>
         new KafkaMessageProducer<>(
             registration.getTopic().getName(), producer, topicProducerCounterSpec, clientId);
 
-    if (micrometerProducerListener != null) {
+    if (micrometerProducerListener == null) {
+      LOGGER.error("MicrometerProducerListener is not initialized! Metrics will not be recorded.");
```

Comment:
NIT: imho this should be `warn`. The service is still functional without the metrics, isn't it?

Reply:
```suggestion
      LOGGER.warn("MicrometerProducerListener is not initialized! Metrics will not be recorded.");
```

Reply:
updated for producer and consumer listener